modules/oidcsso/src/main/java/org/wso2/carbon/tomcat/oidcsso/extension/agent/AuthenticationRequestBuilder.java

```diff
@@ -0,0 +1,120 @@
+/*
+ *  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.carbon.tomcat.oidcsso.extension.agent;
+
+import com.nimbusds.oauth2.sdk.ResponseType;
+import com.nimbusds.oauth2.sdk.Scope;
+import com.nimbusds.oauth2.sdk.id.ClientID;
+import com.nimbusds.oauth2.sdk.id.State;
+import com.nimbusds.openid.connect.sdk.AuthenticationRequest;
+import com.nimbusds.openid.connect.sdk.ClaimsRequest;
+import org.wso2.carbon.tomcat.oidcsso.extension.Constants;
+import org.wso2.carbon.tomcat.oidcsso.extension.oidc.StateStore;
+import org.wso2.carbon.tomcat.oidcsso.extension.utils.exception.AuthenticationRequestException;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A Java class which defines the parameters required to generate an OpenID Connect authentication request.
```

Comment:
(Minor) `"A Java class"` seems a bit redundant, since this is a comment inside a Javadoc. Rather, composition in terms of "This class defines" seems to be better.